src/main/java/com/exasol/projectkeeper/validators/changelog/ChangelogValidator.java

```diff
@@ -0,0 +1,95 @@
+package com.exasol.projectkeeper.validators.changelog;
+
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.project.MavenProject;
+
+import com.exasol.errorreporting.ExaError;
+import com.exasol.projectkeeper.ValidationFinding;
+import com.exasol.projectkeeper.Validator;
+
+/**
+ * Validator that checks the existence of the doc/changes/changes_X.X.X.md file for the current project's version.
+ */
+public class ChangelogValidator implements Validator {
+    public static final String SNAPSHOT_SUFFIX = "-SNAPSHOT";
+    private final MavenProject mavenProject;
+    private final Path relativePathToChangesFile;
+    private final Path changesFileAbsolutePath;
+
+    /**
+     * Create a new instance of {@link ChangelogValidator}
+     * 
+     * @param mavenProject maven project
+     */
+    public ChangelogValidator(final MavenProject mavenProject) {
+        this.mavenProject = mavenProject;
+        this.relativePathToChangesFile = Path.of("doc", "changes", "changes_" + mavenProject.getVersion() + ".md");
+        this.changesFileAbsolutePath = this.mavenProject.getBasedir().toPath().resolve(this.relativePathToChangesFile);
+    }
+
+    @Override
+    public ChangelogValidator validate(final Consumer<ValidationFinding> findingConsumer) {
+        if (!isSnapshotVersion(this.mavenProject.getVersion())) {
+            runValidation(findingConsumer);
+        }
+        return this;
+    }
+
+    private void runValidation(final Consumer<ValidationFinding> findingConsumer) {
```

Comment:
You're again fixing something in the method called validation. At least rename the method

Reply:
but here it's a cool trick ;-)
I take the original file, fix it and then compare. So I use the fixing for validation. I think the method name is correct. this method does validation, but it uses other classes that fix for that.

Reply:
I already noticed that passing the fix as lambda is not the best idea. I will change this in a separate PR since it will affect the whole PK.